docs/DeveloperGuide.md

```diff
@@ -452,6 +447,145 @@ facilitates the lending of money and ensures data integrity by validating input
 The TransactionList class provides essential functionalities for managing a list of transactions.
 Its methods facilitate the addition, removal, editing, and retrieval of transactions, ensuring efficient management of transactions within a group.
 
+### DateTime
+
+<ins> Overview </ins>
+
+The DateTime class handles all operations in LongAh involving the tracking of time. This includes storing and printing
+the datetime elements in dated transactions, parsing user's date & time related inputs as well as filtering transactions
+according to their stored date & time. Implementation of the class is made possible with the help of the *java.time*
+system class.
+
+<ins> Class Fields </ins>
+
+Storing requirements only occurs for the specific datetime component of the class. Hence, the class field structure is
+as follows:
+
+- *dateTime*: A dateTime object from *java.time* representing date & time associated with the current 
+instance.
+
+<ins> Constructor </ins>
+
+The DateTime constructor takes in a string representation of date & time in the `DD-MM-YYYY HHMM` form and parse it into
+a LocalDateTime instance from *java.time* and stores it under the *dateTime* field.
+
+Invalid string date & time inputs to the constructor will trigger exceptions. The exceptions and triggering conditions
+are as follows:
+
+- `INVALID_TIME_FORMAT`: String input representing date & time is not in the `DD-MM-YYYY HHMM` format.
+- `INVALID_TIME_INPUT`: String input is representing a future date & time. This is not permitted in the LongAh system
+  considering real-life practicability.
+
+<ins> Methods </ins>
+
+- *isBefore*: Determines whether an input DateTime object has a dateTime field that is before that of the current
+instance.
+- *isAfter*: Determines whether an input DateTime object has a dateTime field that is after that of the current
+instance.
+- *isEqual*: Determines whether an input DateTime object has a dateTime field that is equal to that of the current 
+instance.
+- *isFuture*: Determines whether the dateTime field of the current instance represent a future date & time (e.g. is 
+after the preset system time). Currently used within the constructor only.
+- *toStorageString*: Formats the dateTime field of the current instance into a String output suitable for loading and 
+storing.
+
+<ins>Usage Example</ins>
+
+The following UML diagram displays how the dateTime component is handled when the user is adding a dated transaction.
+
+![addDateTimeforDatedTransaction.png](diagrams%2FaddDateTimeforDatedTransaction.png)
```

Comment:
Don't include Logger in sequence diagram

Reply:
Factored in LongAhExceptions to keep track of the exceptions instead.

Reply:
Included ref frames to omit non-interested details in class implementation while explaining date time.